lib/trino-record-decoder/src/test/java/io/trino/decoder/DecoderTestColumnHandle.java

```diff
@@ -172,7 +199,8 @@ public boolean equals(Object obj)
                 Objects.equals(this.formatHint, other.formatHint) &&
                 Objects.equals(this.keyDecoder, other.keyDecoder) &&
                 Objects.equals(this.hidden, other.hidden) &&
-                Objects.equals(this.internal, other.internal);
+                Objects.equals(this.internal, other.internal) &&
+                Objects.equals(this.oneOfs, other.oneOfs);
```

Comment:
list equality depends on element order.

Does a protobuf schema consider differently ordered oneofs as different?
At-least within JSON the field order doesn't matter - so maybe this equality method should do the same?

Reply:
Can't say the protobuf schema would treat different ordered `oneof` definitions as different objects. The order of the column names in this list matches the order of how the fields are defined in the schema (by name, not by tag number).  Since order doesn't matter, and the final JSON object will have at most one field, how about we change this to a `Set<String>` instead of `List`? @hashhar

Reply:
Set sounds good if we're only going to keep the single field.